nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -700,28 +714,157 @@ in {
 
     services.nginx = lib.mkIf cfg.configureNginx {
       enable = true;
-      recommendedProxySettings = true; # required for redirections to work
+
+      commonHttpConfig = ''
+        proxy_cache_path /var/cache/nginx levels=1:2 keys_zone=all_cache:10m inactive=7d max_size=1g;
```

Comment:
The `key_zone` variable can have any name, it must be specified in the `proxy_cache` parameter.

Reply:
The name of the zone has no effect on the operation of the cache, it can be anything.

I recently added the proxyCache option to nginx:
```
      proxyCache = {
        enable = true;
        keysZoneName = "all_cache";
        inactive = "7d";
      };
```
I can change it to `keysZoneName = "CACHE";`


But nginx allows you to set several proxy_cache_path parameters:
```
  proxy_cache_path /var/cache/nginx/all levels=1:2 keys_zone=all:10m inactive=7d max_size=1g;
  proxy_cache_path /var/cache/nginx/mastodon levels=1:2 keys_zone=mastodon:10m inactive=7d max_size=1g;
```

I want to redo the `nginx.proxyCache` option and make the configuration look like this:
```
      proxyCache = {
        "mastodon" = {
          enable = true;
          keysZoneName = "mastodon";
          inactive = "7d";
        };
      };
```

In the process, I encountered errors that I don't know how to correct.